src/hashi/HashiProverLib.sol

```diff
@@ -0,0 +1,248 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.0;
```

Comment:
This contract all looks good but I'd need to review in the context of how it fits in the wider application and the workflow. Would appreciate a walkthrough.